handler/oauth2/flow_refresh.go

```diff
@@ -96,13 +95,39 @@ func (c *RefreshTokenGrantHandler) HandleTokenEndpointRequest(ctx context.Contex
 	}
 
 	request.SetSession(originalRequest.GetSession().Clone())
-	request.SetRequestedScopes(originalRequest.GetRequestedScopes())
+
+	/*
+			There are two key points in the following spec section this addresses:
+				1. If omitted the scope param should be treated as the same as the scope originally granted by the resource owner.
+				2. The REQUESTED scope MUST NOT include any scope not originally granted.
+
+			scope
+					OPTIONAL.  The scope of the access request as described by Section 3.3.  The requested scope MUST NOT
+		  			include any scope not originally granted by the resource owner, and if omitted is treated as equal to
+		   			the scope originally granted by the resource owner.
+
+			See https://www.rfc-editor.org/rfc/rfc6749#section-6
+	*/
+	switch scope := request.GetRequestForm().Get("scope"); scope {
+	case "":
+		// Addresses point 1 of the text in RFC6749 Section 6.
+		request.SetRequestedScopes(originalRequest.GetGrantedScopes())
+	default:
+		request.SetRequestedScopes(fosite.RemoveEmpty(strings.Split(scope, " ")))
```

Comment:
Would this also work if the scope query parameter is `?scope=foo+bar`?

Reply:
Let me double check. I believe I copied this specific line from elsewhere.

Reply:
This is the source which should theoretically mean the issue exists here too: https://github.com/ory/fosite/blob/ef3e950835e8dd0ab3405f36745b962ede11e3a8/access_request_handler.go#L80

It appears the go http package parses this correctly as far as I can see locally, Both url.ParseQuery  and url.ParseRequestURI seem to treat the `+` as a space, similar to `%20`.